src/redis/cache_impl.go

```diff
@@ -16,7 +16,12 @@ import (
 )
 
 type rateLimitCacheImpl struct {
-	pool                       Pool
+	pool Pool
+	// Optional Pool for a dedicated cache of per second limits.
+	// If this pool is nil, then the Cache will use the pool for all
+	// limits regardless of unit. If this pool is not nil, then it
+	// is used for limits that have a SECOND unit.
```

Comment:
if i were to set a 3600 second limit it would end up in the per second pool. is that correct?

Reply:
Do you meant a limit of 3600 with a SECOND unit, or a limit that has an expiration of 3600 seconds?

I think you mean the latter. You are right that the current code would need to be modified to handle cases like that, but the nice thing is that we currently don't support that type of unit semantics. So you *can't* specify a limit with an expiration of 3600 seconds.

Reply:
Ah right.